spec/models/miq_provision_spec.rb

```diff
@@ -78,11 +80,37 @@
           expect(@vm_prov.get_option(:vm_target_hostname)).to eq(name_002.gsub(/ +|_+/, "-"))
         end
 
+        context "#update_vm_name" do
+          it "does not modify a fully resolved vm_name" do
+            @vm_prov.update_vm_name(@target_vm_name)
+            expect(@vm_prov.get_option(:vm_target_name)).to eq(@target_vm_name)
+          end
+
+          it "Enumerates vm_name that contains the naming sequence characters" do
+            expect(MiqRegion).to receive(:my_region).and_return(miq_region)
+
+            @vm_prov.update_vm_name("#{@target_vm_name}$n{3}")
+            expect(@vm_prov.get_option(:vm_target_name)).to eq("#{@target_vm_name}001")
+          end
+
+          it "Updates the request description with only name parameter passed" do
+            expect(@pr).to receive(:update_description_from_tasks)
```

Comment:
Should there be more to this than just checking that the method was called?

Reply:
Yup, working on it.